intro-to-tests/practice-intro-to-pytest.md

```diff
@@ -0,0 +1,910 @@
+        return "Type Error: num_a and num_b must be integer or float"
+    
+    operator = operator.lower()
+    if operator in ["add", "+"]:
+        return num_a + num_b
+    elif operator in ["subtract", "-"]:
+        return num_a - num_b
+    elif operator in ["multiply", "*"]:
+        return num_a * num_b
+    elif operator in ["divide", "/"]:
+        if num_b == 0:
+            return "Zero Division Error"
+        else:
+            return num_a / num_b
+    else:
+        return "Value Error: Operator Not Found"
+```
+
+Consider `calculate()`. Which of these are valid test cases to have assertions for?
+
+##### !end-question
+##### !options
+
+* When `num_a` is not an integer or float, returns `"Type Error"`
+* When `num_b` is not an integer or float, returns `"Type Error"`
+* Successful addition when operator is `"add"`
+* Successful addition when operator is `"ADD"`
+* Successful addition when operator is `"+"`
+* Successful subtraction when operator is `"subtract"`
+* Successful subtraction when operator is `"SUBTRACT"`
+* Successful subtraction when operator is `"-"`
+* Successful multiplication when operator is `"multiply"`
+* Successful multiplication when operator is `"MULTIPLY"`
+* Successful multiplication when operator is `"*"`
+* Successful division when operator is `"divide"`
+* Successful division when operator is `"DIVIDE"`
+* Successful division when operator is `"/"`
+* When `num_b` is `0` and operator is division, returns `"Zero Division Error"`
+* When `operator` is an invalid value, such as `"elephant"` or `True`, returns `"Value Error: Operator Not Found"`
+
+##### !end-options
+##### !answer
+
+* When `num_a` is not an integer or float, returns `"Type Error"`
+* When `num_b` is not an integer or float, returns `"Type Error"`
+* Successful addition when operator is `"add"`
+* Successful addition when operator is `"ADD"`
+* Successful addition when operator is `"+"`
+* Successful subtraction when operator is `"subtract"`
+* Successful subtraction when operator is `"SUBTRACT"`
+* Successful subtraction when operator is `"-"`
+* Successful multiplication when operator is `"multiply"`
+* Successful multiplication when operator is `"MULTIPLY"`
+* Successful multiplication when operator is `"*"`
+* Successful division when operator is `"divide"`
+* Successful division when operator is `"DIVIDE"`
+* Successful division when operator is `"/"`
+* When `num_b` is `0` and operator is division, returns `"Zero Division Error"`
+* When `operator` is an invalid value, such as `"elephant"` or `True`, returns `"Value Error: Operator Not Found"`
+
+##### !end-answer
+##### !hint
+
+The best test suites have a test case to cover every meaningful combination of inputs and their expected outputs.
+
+##### !end-hint
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 6 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: ordering
+* id: 0liBZs
+* title: Intro to Pytest
+##### !question
+
```

Comment:
whoa... I didn't know there were ordering type questions in learn!?!? I don't see that in the cli!? : O @tildeee

Reply:
JASMINE TAUGHT ME! IT'S EXTREMELY COOL!